bytecomp/simplif.ml

```diff
@@ -282,7 +286,9 @@ let simplify_exits lam =
             xs ys Ident.Map.empty
         in
         List.fold_right2
-          (fun (y, kind) l r -> Llet (Alias, kind, y, l, r))
+          (fun (y, kind) l r ->
+             Llet ((if can_alias l then Alias else Strict), kind, y, l, r)
```

Comment:
I would still prefer `Llet (Struct, kind, y, l, r)`, as the obviously-correct bug fix.

Note that in bytecode debug mode, if an `Alias` is generated and `y` happens to be a user-visible variable, it won't be visible in the debugger.

Reply:
But that was the behavior in 4.07, right?  I'm really concerned about introducing a major change so close from the release.  But I won't ask you to insist even more, I'll push the simpler (and more risky, IMO) version.

Reply:
Sorry, I'm having second thoughts.  I really don't think it is right to fix the bug by changing all Alias introduced by simplify_exits into Strict so close from a release.  I'll instead push a fix that disables the new optimization.

Reply:
I haven't followed the tribulations of this code, but in 4.07 it was something else again:
```
        List.fold_right2
          (fun y l r -> Llet (Alias, Pgenval, y, l, r))
          ys ls (Lambda.subst env handler)
```

Reply:
It's essentially the same, without the "kind" which is new.  But it was definitely an `Alias` (which was morally wrong, but probably harmless).